src/components/VmDetails/cards/SnapshotsCard/SnapshotItem.js

```diff
@@ -34,31 +47,62 @@ StatusTooltip.propTypes = {
   id: PropTypes.string.isRequired,
 }
 
-function isSnapshotBeingDeleted (snapshotId, pendingTasks) {
+function isSnapshotDeleted (snapshotId, pendingTasks) {
   return !!pendingTasks.find(task => task.type === PendingTaskTypes.SNAPSHOT_REMOVAL && task.snapshotId === snapshotId)
 }
 
 const SnapshotItem = ({ snapshot, vmId, isEditing, beingDeleted, onSnapshotDelete }) => {
-  let buttons = []
   let statusIcon = null
+  let buttons = []
+
+  // Snapshot actions
+  const isActionsDisabled = beingDeleted || !isEditing || snapshot.get('status') === 'locked'
   if (!snapshot.get('isActive')) {
+    // Info popover
     buttons.push(<OverlayTrigger
-      overlay={<SnapshotDetail key='detail' snapshot={snapshot} vmId={vmId} />}
+      overlay={
+        <SnapshotDetail key='detail' snapshot={snapshot} vmId={vmId} restoreDisabled={isActionsDisabled} />
+      }
       placement='left'
       trigger='click'
       rootClose
       key='info'
     >
-      <a><Icon type='pf' name='info' /></a>
+      <a>
+        <Icon type='pf' name='info' />
+      </a>
     </OverlayTrigger>)
-    if (isEditing && !snapshot.get('isActive') && !beingDeleted) {
-      buttons.push(<RestoreConfirmationModal key='restore' snapshot={snapshot} vmId={vmId} />)
-      buttons.push(<DeleteConfirmationModal key='delete' trigger={<a key='delete'><Icon type='pf' name='delete' /></a>} onDelete={onSnapshotDelete}>
+
+    // Delete and restore actions
+    if (!snapshot.get('isActive')) {
```

Comment:
Looks like this if condition is repeating the if condition above.  If so, this `isActive` check isn't necessary.

Reply:
I think the check was moved up to `isActionsDisabled` and this `if` isn't needed anymore.  The icons will show but be disabled.